parser/Parser.py

```diff
@@ -0,0 +1,133 @@
+from neo4j import GraphDatabase
+import re,unicodedata,unidecode
+
+
+class Parser:
+
+    def __init__(self,filename,uri="bolt://localhost:7687"):
+        self.driver = GraphDatabase.driver(uri)
+        self.session = self.driver.session()
+        self.filename=filename
+
+    def openfile(self):
+        """open the file filename and return the list of entries, each entry is a list a the sanitized lines"""
+        name = self.filename + ( '.txt' if (len(self.filename)<4 or self.filename[-4:]!=".txt") else '')  #to not get error if extension is missing
+        txtfile=open(name,"r",encoding='utf8')
+        file = [ [ ] ]
+        for line in txtfile:
+            if line=='\n' :
+                if len(file[-1])>0:
+                    file.append([])
+            else:
+                #sanitizing the line before adding it
+                line = line.rstrip()
+                line = line.replace("’", "'")
+                line = re.sub(r"(\d),(\d)", r"\1‚\2", line)
+                line = re.sub(r"\\,", "\\‚", line)
+                line = re.sub(r"\(([ivx]+)\)", r"\1", line, flags=re.I)
+                file[-1].append(line)
+        self.file = file
+
+
+    def harvest(self):
+        """Transform data in file to nodes in with neo4j"""
+        def normalizing(line,lang="default"):
+            """normalize a string depending of the language code lang"""
+            line = unicodedata.normalize("NFC", line)
+            if lang in ['fr','ca','es','it','nl','pt','sk','en']: #removing accent
+                line = unidecode.unidecode(line)
+            if lang not in [ ]: #lower case except if language in list
+                line=line.lower()
+            line = re.sub(r"[\u0000-\u0027\u200b]", "-", line)
+            line = re.sub(r"&\w+;", "-", line)
+            line = re.sub(r"[\s!\"#\$%&'()*+,\/:;<=>?@\[\\\]^_`{\|}~¡¢£¤¥¦§¨©ª«¬®¯°±²³´µ¶·¸¹º»¼½¾¿×ˆ˜–—‘’‚“”„†‡•…‰‹›€™\t]", "-", line)
+            line = re.sub(r"-+", "-", line)
+            line = line.strip("-")
+            return line
+
+        def add_line(line):
+            """to get a normalized string but keeping the language code "lc:" and the "," separators"""
+            if len(line)<=3:return line
+            else:
+                lang=line[:2]
+                new_line=lang+":"
+                for word in line[3:].split(","):
+                    new_line+=normalizing(word,lang)+','
+                new_line=new_line[:-1]
+                return new_line
+
+        index_comm=0
+        index_stopwords=1
+        index_synonyms=1
+        stop = re.compile('stopword')
+        syno = re.compile('synonym')
+        lc = re.compile('[a-zA-Z][a-zA-Z]:')
+
+        #header
+        h=0
+        header=''
+        while self.file[h][0][0]=="#":
+            for line in self.file[h]:
+                header+=line
+            h+=1
+        query="CREATE (node{id:'__header__'" + (",value:"+header if len(header)>0 else "") +"})"
```

Comment:
it would be better to have sub method for writing to database, it will really ease unit testing but also it's cleaner.